pypsa/plot.py

```diff
@@ -247,27 +256,40 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
         bus_collection = ax.scatter(x, y, c=c, s=s, cmap=bus_cmap, edgecolor='face')
 
     def as_branch_series(ser):
+        # ensure that this function always return a multiindexed series
         if isinstance(ser, dict) and set(ser).issubset(branch_components):
-            return pd.Series(ser)
-        elif isinstance(ser, pd.Series):
-            if isinstance(ser.index, pd.MultiIndex):
-                return ser
-            index = ser.index
-            ser = ser.values
+            return network.branches().loc[branch_components] \
+                          .join(pd.Series(ser, name='value') \
+                          .rename_axis('component')).value
+        elif isinstance(ser, pd.Series) and isinstance(ser.index, pd.MultiIndex):
+                return ser.rename_axis(['component', 'name'])
         else:
-            index = network.lines.index
-        return pd.Series(ser,
-                         index=pd.MultiIndex(levels=(["Line"], index),
-                                             codes=(np.zeros(len(index)),
-                                                    np.arange(len(index)))))
+            ser =  pd.Series(ser, network.lines.index)
+        return pd.concat([ser], axis=0, keys=['Line'],
+                         names=['component', 'name']).fillna(0)
 
     line_colors = as_branch_series(line_colors)
     line_widths = as_branch_series(line_widths)
+
     if not isinstance(line_cmap, dict):
         line_cmap = {'Line': line_cmap}
 
     branch_collections = []
 
+    if flow is not None:
+        flow = _flow_ds_from_arg(flow, network, branch_components)\
+                        .pipe(as_branch_series)\
+                        .div(len(network.branches()) + 100)\
+                        .mul(line_widths, fill_value=1)
```

Comment:
Please prefer extra `(` to `)` to line continuations with `\`. Not all people are used to them, often the indentation is all over the place and its easy to miss one.

```python
         flow = (_flow_ds_from_arg(flow, network, branch_components)
                 .pipe(as_branch_series)
                 .div(len(network.branches()) + 100)
                 .mul(line_widths, fill_value=1))
```

```
network.branches()
```
is to be considered expensive.

```python
sum(len(t.df) for t in network.iterate_components(branch_components))
```

Reply:
yes, agreed! Thanks for revising